src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -25,6 +25,7 @@ public class JsonRpc : IDisposableObservable
     {
         private const string ImpliedMethodNameAsyncSuffix = "Async";
         private const string CancelRequestSpecialMethod = "$/cancelRequest";
```

Comment:
>CancelRequestSpecialMethod [](start = 29, length = 26)

One thing we need to talk about is how will cancellation work with progress?

Reply:
What issues do you anticipate, @tinali@microsoft.com ? The server can send `$/progress` and the client can send `$/cancelRequest`. The `$/progress` is valid until the server sends a result or error back. Cancel doesn't change that requirement: the server must still send the response back.

---
In reply to: [304607745](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304607745) [](ancestors = 304607745)